### PR TITLE
feat: add frequency support for lr_scheduler [DET-5021]

### PR DESCRIPTION
**Pull request by hamidzr**
## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: harness/determined/pytorch/_lr_scheduler.py [hunk 1]
```diff
@@ -43,10 +44,10 @@ def __init__(
                 The strategy Determined will use to call (or not call) scheduler.step().
 
                 1. ``STEP_EVERY_EPOCH``: Determined will call scheduler.step() after
-                   every training epoch. No arguments will be passed to step().
+                   every ``frequency`` training epoch. No arguments will be passed to step().
```
Comment (shiyuann) [inline harness/determined/pytorch/_lr_scheduler.py:47]:
```suggestion
                   every ``frequency`` training epoch(s). No arguments will be passed to step().
```

File: harness/determined/pytorch/_lr_scheduler.py [hunk 2]
```diff
@@ -43,10 +44,10 @@ def __init__(
...
 
                 2. ``STEP_EVERY_BATCH``: Determined will call scheduler.step() after every
-                   training batch. No arguments will be passed to step().
+                   ``frequency`` training batch. No arguments will be passed to step().
```
Comment (shiyuann) [inline harness/determined/pytorch/_lr_scheduler.py:50]:
```suggestion
                   ``frequency`` training batch(es). No arguments will be passed to step().
```

File: harness/determined/pytorch/_pytorch_trial.py [hunk 3]
```diff
@@ -253,9 +253,10 @@ def _auto_step_lr_scheduler_per_batch(
         This function aims at automatically step a LR scheduler. It should be called per batch.
         """
         if lr_scheduler._step_mode == pytorch.LRScheduler.StepMode.STEP_EVERY_BATCH:
-            lr_scheduler.step()
+            if batch_idx % lr_scheduler._frequency == 0:
```
Comment (shiyuann) [inline harness/determined/pytorch/_pytorch_trial.py:256]: `batch_idx` starts from 0 and it seems to always step when it begins training. You should use `batch_idx + 1` instead.

**Comment by shiyuann** [comment]
@hamidzr please also add a release note for the Pytorch features and the Pytorch Lightning features.

**Comment by hamidzr** [comment]
> @hamidzr please also add a release note for the Pytorch features and the Pytorch Lightning features.

I added the release note for pytorch feature added in this pr. I'll add the lightning release notes to #2078 how does that sound?

**Review by shiyuann — APPROVED** [review]
LGTM!
